### ISSUE TITLE
reitit.ring/create-resource-handler :not-found-handler is not called 

### ISSUE THREAD

**Issue by paulbutcher** [metosin/reitit#464]
I’m unable to get the `:not-found-handler` option for `reitit.ring/create-resource-handler` to work: it doesn’t look like the function is called at all.

I’ve created a minimal cut-down version of my application which demostrates the problem: https://github.com/paulbutcher/reitit-bug

The relevant code is:

```clojure
(def handler
  (ring/ring-handler
   (ring/router
    ["/ping" {:get (constantly (response "pong"))}])
   (ring/routes
    (ring/create-resource-handler {:path "/"
                                   :not-found-handler (fn [x]
                                                        (println "not found handler: " x)
                                                        (-> "index.html"
                                                            (resource-response {:root "public"})
                                                            (content-type "text/html")))})
    (ring/create-default-handler))))
```

**Comment by miikka** [comment]
The not found handler only works in the case that `:path` has not been set as can seen from the implementation, linked below. I'm not sure about the rationale behind the implementation.

https://github.com/metosin/reitit/blob/0ecc2fb9a7c5d93f2bb7582fbdedf75e8dc8e484/modules/reitit-ring/src/reitit/ring.cljc#L221-L230

My guess is that it was left out because `:not-found-handler` defaults to a 404 handler and not to `(constantly nil)`, preventing the handler from composing the usual way (which is what you want when `:path` is set). It think it could make sense to change it so that if `:path` matches but the resource is not found, then the not found handler is called.

To make the change backwards compatible, the default value for `:not-found-handler` would need to be set to `(constantly nil)` in case `:path` is set.

**Comment by miikka** [comment]
Will be fixed in the next release. https://github.com/metosin/reitit/pull/473 needs to resolved first.

**Comment by miikka** [comment]
Released in 0.5.13.
